src/particles/particle_creation.hpp

```diff
@@ -135,6 +139,242 @@ template <ParticleType particleType> struct ParticleCreationTraits {
 	}
 };
 
+#if AMREX_SPACEDIM == 3
+// Specialization for StochasticStellarPop particles
+template <> struct ParticleCreationTraits<ParticleType::StochasticStellarPop> {
+	// Specialized nested ParticleChecker for StochasticStellarPop particles
+
+	static constexpr amrex::Real eps_star = 0.5; // fraction of gas mass that goes into star particles
+	static constexpr amrex::Real eps_ff = 0.5;   // efficiency per free fall time
+	static constexpr amrex::Real J = 0.5;	     // Jeans parameter
+
+	// Constants for the Chabrier IMF
+	static constexpr amrex::Real m_star_high = 8.0 * C::M_solar; // all stars above this mass are considered high mass stars
+	static constexpr amrex::Real m_imf_min = 0.08 * C::M_solar;  // lower limit of the IMF
+	static constexpr amrex::Real m_imf_max = 120.0 * C::M_solar; // high mass limit of the IMF
+	static constexpr amrex::Real m_imf_break = 8.0 * C::M_solar; // IMF is lognormal below this mass and powerlaw above
+	static constexpr amrex::Real imf_disp = 0.55;		     // dispersion of the lognormal IMF
+	static constexpr amrex::Real imf_mu = 32.599;		     //=log10(0.2 * C::M_solar), mean of the lognormal IMF, avoid compiler error
+	static constexpr amrex::Real alpha = 2.35;		     // slope of the powerlaw
+
+	static double fstar_high;      // fstar is the fraction of number of high mass stars from the IMF
+	static double m_star_high_avg; // average mass of high mass stars
+
+	ParticleCreationTraits()
+	{
+		auto arg = [](double mass) -> double { return (std::log10(mass) - imf_mu) / std::sqrt(2.0 * imf_disp * imf_disp); };
+		double const norm_ratio =
+		    std::pow(m_imf_break, (1 - alpha)) * imf_disp * std::sqrt(2.0 * M_PI) / std::exp(-arg(m_imf_break) * arg(m_imf_break));
+
+		double const total_stars = (1. - alpha) * norm_ratio * (std::erf(arg(m_imf_break)) - std::erf(arg(m_imf_min))) +
+					   std::pow(m_imf_max, 1.0 - alpha) - std::pow(m_imf_break, 1.0 - alpha);
+		double const num_high_mass_stars = std::pow(m_imf_max, 1.0 - alpha) - std::pow(m_star_high, 1.0 - alpha);
+		;
+
+		fstar_high = num_high_mass_stars / total_stars;
+		m_star_high_avg = m_imf_max * ((alpha - 1.0) / (alpha - 2.0)) * (1. - std::pow(m_star_high / m_imf_max, 2.0 - alpha)) /
+				  (1. - std::pow(m_star_high / m_imf_max, 1.0 - alpha));
+	}
+
+	template <typename problem_t> struct ParticleChecker {
+		amrex::Real current_time;
+		amrex::Real dt;
+		amrex::Real param1 = particle_param1;
+		amrex::Real param2 = particle_param2;
+
+		AMREX_GPU_HOST_DEVICE ParticleChecker(amrex::Real current_time, amrex::Real dt) : current_time(current_time), dt(dt) {}
+
+		AMREX_GPU_DEVICE auto operator()(amrex::Array4<const amrex::Real> const &state_arr, int i, int j, int k,
+						 amrex::GpuArray<amrex::Real, AMREX_SPACEDIM> const &dx) const -> int
+		{
+			// Estimate the Jeans length(LambaJ) and create particle if LambaJ < J*dx
```

Comment:
This comment should probably be changed, since it looks like it dates from before creation was stochastic.